pkg/sfu/buffer/rtpstats_sender.go

```diff
@@ -647,55 +627,58 @@ func (r *RTPStatsSender) GetExpectedRTPTimestamp(at time.Time) (expectedTSExt ui
 	return
 }
 
-func (r *RTPStatsSender) GetRtcpSenderReport(ssrc uint32) *rtcp.SenderReport {
+func (r *RTPStatsSender) GetRtcpSenderReport(ssrc uint32, publisherSRData *RTCPSenderReportData, tsOffset uint64) *rtcp.SenderReport {
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
-	if !r.initialized {
+	if !r.initialized || publisherSRData == nil {
 		return nil
 	}
 
-	// construct current time based on monotonic clock
-	timeSinceFirst := time.Since(r.firstTime)
-	if timeSinceFirst < cSenderReportInitialWait {
-		return nil
-	}
-	now := r.firstTime.Add(timeSinceFirst)
+	timeSincePublisherSR := time.Since(publisherSRData.At)
+	now := publisherSRData.At.Add(timeSincePublisherSR)
 	nowNTP := mediatransportutil.ToNtpTime(now)
-	nowRTPExt := r.extStartTS + uint64(timeSinceFirst.Nanoseconds()*int64(r.params.ClockRate)/1e9)
+	nowRTPExt := publisherSRData.RTPTimestampExt - tsOffset + uint64(timeSincePublisherSR.Nanoseconds()*int64(r.params.ClockRate)/1e9)
```

Comment:
This is where the subscriber sender report. It is based on the latest publisher sender report. Note that we are using the clock rate to account for time elapsed since the publisher sender report was received. If publisher is actively changing clock rate (like we have seen with some noise suppression cases), it will not be accounted for the duration we are using clock rate here for. But, there is no better option.

One option is to make RTCP report forwarding completely event driven. Currently, there is a worker which is collecting sender reports for all subscribed tracks every three seconds and batching a bunch of reports in the same packet. Fully event driven will break up the packet a lot.